### PR TITLE
added authorization

### PR DESCRIPTION
**Pull request by DavidHewWing**
Added authorization

Testing:
1. Get key from Slack Backend Channel and `yarn dev`
2. Test a route (v0/contests/status) with header: 'auth-token': <key>.
3. Put the valid key and invalid key

### CODE REVIEW

File: src/routes/api/contest.route.js [hunk 1]
```diff
@@ -1,7 +1,8 @@
 const express = require('express');
 const router = express.Router()
```
Comment (Wuon) [inline src/routes/api/contest.route.js]: move this below require statements. its better to group them at the top like importing packages in java

Comment (ZGao28) [reply src/routes/api/contest.route.js]: interesting that the linter didn't catch this

Comment (Wuon) [reply src/routes/api/contest.route.js]: eslint isn't configured properly

File: src/routes/api/contest.route.js [hunk 2]
```diff
@@ -1,7 +1,8 @@
...
 const contestController = require('../../controllers/contest.controller')
+const auth = require('../../middlewares/authorization')
 
-router.get('/status', contestController.status);
+router.get('/status', auth, contestController.status);
```
Comment (Wuon) [inline src/routes/api/contest.route.js]: This makes me believe this endpoint is a health check of some sort, but upon investigation, the function is something else. I suggest renaming it to something more appropriate that follows a naming convention

Comment (Wuon) [reply src/routes/api/contest.route.js]: GET/v0/contests
GET/v0/contests/:contestId
GET/v0/rankings

File: src/routes/api/contest.route.js (hunk 1 above)
Comment (callmekungfu) [reply src/routes/api/contest.route.js]: You need to add the eslint extension that checks imports

it's called eslint-plugin-import

https://www.npmjs.com/package/eslint-plugin-import

File: src/routes/api/contest.route.js (hunk 2 above)
Comment (DavidHewWing) [reply src/routes/api/contest.route.js]: These status endpoints for **contests** and **users**, are placeholder for the time being. 
If you hit **GET/v0/contests/status** and **GET/v0/users/status**.

I have already configured the API routes to that naming convention in this PR: https://github.com/uosoftwareclub/core-service/pull/1

**Comment by DavidHewWing** [comment]
@ZGao28 We are using keys, that I have stored in the Mongo Collection.

File: src/middlewares/authorization.js [hunk 3]
```diff
@@ -1,3 +1,19 @@
...
+  if (!token) return res.status(401).send('Access Denied');
+  
+  Key.findOne({ 'key' : token}, 'read write', (err, key) => {
+    if (err) return errorHandler.handleError(err);
+    if (!key) return res.status(400).send('Invalid Token');
+
+    const { read, write } = key;
+    req.read = read;
+    req.write = write;
+    next()
```
[the remark this answers is not in the document]
Comment (DavidHewWing) [reply src/middlewares/authorization.js]: next causes time out.

**Comment by Wuon** [comment]
Can you create the issues @DavidHewWing for the following suggestions in this PR?

**Comment by DavidHewWing** [comment]
@Wuon Yessir

**Pull request merged**